File: frontend/src/types/employee.ts
export interface Employee {
  id: string;
  employee_id: string;
  full_name: string;
  email: string;
  department: string;
  created_at: string;
}

export interface EmployeeCreate {
  full_name: string;
  email: string;
  department: string;
}

File: backend/app/models/employee.py
from pydantic import BaseModel, EmailStr, Field
from datetime import datetime


class EmployeeCreate(BaseModel):
    employee_id: str = Field(..., example="EMP001")
    full_name: str
    email: EmailStr
    department: str


class EmployeeResponse(EmployeeCreate):
    id: str
    created_at: datetime

class NewEmployeeCreation(BaseModel):
    full_name: str
    email: EmailStr
    department: str